apiserver/server_test.go

```diff
@@ -512,6 +512,64 @@ func (s *serverSuite) TestAPIHandlerConnectedModel(c *gc.C) {
 	c.Check(handler.ConnectedModel(), gc.Equals, otherState.ModelUUID())
 }
 
+func (s *serverSuite) TestClosesStateFromPool(c *gc.C) {
+	pool := state.NewStatePool(s.State)
+	server := newServer(c, s.State, pool)
+	defer server.Stop()
+
+	otherState := s.Factory.MakeModel(c, nil)
+	defer otherState.Close()
+
+	s.State.StartSync()
+	// This ensures that the model exists for more than one of the
+	// time slices that the watcher uses for coalescing
+	// events. Without it the model appears and disappears quickly
+	// enough that it never generates a change from WatchModels.
+	// Many Bothans died to bring us this information.
+	time.Sleep(coretesting.ShortWait)
```

Comment:
Would it be more reliable to have the test have it's own model watcher from the same State instance that the apiserver is using? It could then watch for the new model to appear itself. I'm worried on a heavily loaded test machine, the sleep might not be enough.